### PR TITLE
Fix/implement block quote design

### LINKED ISSUE #184: Implement the design of the quote block
Current design: 
quote_

In comparison to the pull quote these blocks only have one width and no color settings.
Related files:
https://github.com/WordPress/twentytwentyone/tree/trunk/assets/sass/05-blocks/quote

### PR DESCRIPTION
**Pull request by JustinyAhin**
Fixes #184.

## Relevant technical choices:
No.

## Test instructions

Add two block quotes in the editor
In frontend, they should look like the model in this figma file https://www.figma.com/file/OEtsCj8oD8JPVx4OvsS0U2/Twenty-Twenty-One?node-id=1254%3A0.

## Screenshots

Quote block in the editor before
block-quote-editor-before

Quote block in the editor after
block-quote-editor-after

Quote block in frontend before
block-quote-frontend-before

Quote block in frontend after
block-quote-frontend-after

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

**Review by melchoyce — CHANGES_REQUESTED** [review]
Hey @JustinyAhin! Thanks for taking this on.

I'm seeing a couple design issues:

1. The alignment of the quotes isn't inside the content width in the editor:

image

2. The quotation marks should show up inside the editor, like the pullquote block:

image

3. The quotation mark spacing on the front-end isn't right:

image

The upper edge of the quotation mark should align to the top of the text, and should be 20px to the left of the text.

Let me know if you have any questions!

**Comment by JustinyAhin** [comment]
Thanks for looking @melchoyce .
Working on your changes request.

**Comment by JustinyAhin** [comment]
> The alignment of the quotes isn't inside the content width in the editor:

This is not related to the new implemented design. The quote block is initially not aligned with the content width. I think this should be solved in another issue.

Screenshot_2020-10-14 Edit Post ‹ trunk wordpress test — WordPress

File: assets/sass/01-settings/global.scss [hunk 1]
```diff
@@ -224,11 +224,11 @@ $baseline-unit: 10px;
 	--quote--border-width: 1px;
 	--quote--font-family: var(--global--font-secondary);
 	--quote--font-size: var(--global--font-size-md);
-	--quote--font-size-large: var(--global--font-size-lg);
+	--quote--font-size-large: 2rem;
```
Comment (rolf-yoast) [inline assets/sass/01-settings/global.scss]: Why did you change it to rem and not edit the variable?

Comment (JustinyAhin) [reply assets/sass/01-settings/global.scss]: @rolf-yoast because `--global--font-size-lg` is used elsewhere.
I just made a commit that added a new `--global--font-size-llg`

Comment (melchoyce) [reply assets/sass/01-settings/global.scss]: Hey @JustinyAhin, instead of making a new variable, how about using the `calc` CSS function with an existing variable?

Comment (JustinyAhin) [reply assets/sass/01-settings/global.scss]: @melchoyce just sent a new commit.

File: assets/sass/01-settings/global.scss [hunk 2]
```diff
@@ -224,11 +224,11 @@ $baseline-unit: 10px;
 	--quote--border-width: 1px;
 	--quote--font-family: var(--global--font-secondary);
 	--quote--font-size: var(--global--font-size-md);
-	--quote--font-size-large: var(--global--font-size-lg);
+	--quote--font-size-large: calc(.5 * var(--global--font-size-xxl));
```
Comment (ryelle) [inline assets/sass/01-settings/global.scss]: You're basing this on the `--global--font-size-xxl` which actually bumps up to 6rem on larger screens, and I don't _think_ we want that on large quotes, it seems too large.

Screen Shot 2020-10-15 at 11 17 38 AM

Maybe this?

```suggestion
	--quote--font-size-large: calc(1.6 * var(--global--font-size-base));
```

File: assets/sass/05-blocks/quote/_editor.scss [hunk 3]
```diff
@@ -1,19 +1,52 @@
...
 
+	&::before {
+		color: currentColor;
+		content: "\201C";
+		display: block;
+		font-size: 3rem;
+		font-weight: 700;
+		line-height: 1;
+		position: absolute;
+		top: 4px;
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_editor.scss]: I'm not sure if it's just my fonts/browser, but this lines up better with `top: 0`

Screen Shot 2020-10-15 at 11 23 36 AM

File: assets/sass/05-blocks/quote/_editor.scss [hunk 4]
```diff
@@ -26,36 +59,16 @@
...
-		.has-background:not(.has-background-background-color) &,
-		[class*="background-color"]:not(.has-background-background-color) &,
-		[style*="background-color"]:not(.has-background-background-color) &,
-		.wp-block-cover[style*="background-image"] & {
-			color: currentColor;
+		&::before {
+			font-size: 5rem;
+			position: absolute;
+			top: 2px;
+			left: -20px;
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_editor.scss]: Same note as above with `top: 0;`, and if you make that change these properties can be removed, since it inherits from the above `.wp-block-quote::before`.

File: assets/sass/05-blocks/quote/_style.scss [hunk 5]
```diff
@@ -16,18 +16,40 @@
...
+		font-size: 3rem;
+		font-weight: 700;
+		line-height: 1;
+		position: absolute;
+		top: 4px;
+		left: -20px;
+	}
+
 	p {
+		color: var(--global--color-primary);
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_style.scss]: Why is this needed?

File: assets/sass/05-blocks/quote/_style.scss [hunk 6]
```diff
@@ -1,7 +1,7 @@
 .wp-block-quote {
-	border-left: var(--quote--border-width) solid var(--quote--border-color);
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_style.scss:2]: I think we're removing all uses of these two variables, so they can be removed from `assets/sass/01-settings/global.scss` too.

File: assets/sass/05-blocks/quote/_style.scss [hunk 7]
```diff
@@ -77,15 +93,15 @@
...
-	.has-background:not(.has-background-background-color) &,
-	[class*="background-color"]:not(.has-background-background-color) &,
-	[style*="background-color"] &,
-	.wp-block-cover[style*="background-image"] & {
-		border-color: currentColor;
+		&::before {
+			font-size: 5rem;
+			position: absolute;
+			top: 2px;
+			left: -20px;
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_style.scss]: Same comment as on `_editor.scss`, you can probably remove these rules.

File: assets/sass/05-blocks/quote/_style.scss [hunk 8]
```diff
@@ -16,18 +16,40 @@
...
 
+	&::before {
+		color: currentColor;
+		content: "\201C";
+		display: block;
+		font-size: 3rem;
+		font-weight: 700;
+		line-height: 1;
+		position: absolute;
+		top: 4px;
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_style.scss]: Same as `_editor.scss`, this lines up better with `top: 0`.

File: assets/sass/05-blocks/quote/_style.scss [hunk 9]
```diff
@@ -16,18 +16,40 @@
...
+		font-weight: 700;
 	}
 
 	.wp-block-quote__citation,
 	cite,
 	footer {
 		color: var(--global--color-primary);
-		font-size: var(--global--font-size-xs);
+		font-size: var(--global--font-size-sm);
+		font-style: var(--pullquote--font-style);
```
Comment (ryelle) [inline assets/sass/05-blocks/quote/_style.scss]: Did you mean for this to be `--quote--font-style`? Same issue in `_editor.scss`.
```suggestion
		font-style: var(--quote--font-style);
```

**Review by ryelle — CHANGES_REQUESTED** [review]

File: assets/sass/05-blocks/quote/_style.scss (hunk 5 above)
Comment (JustinyAhin) [reply assets/sass/05-blocks/quote/_style.scss]: Oh, this is also inherited. Removing it.

File: assets/sass/05-blocks/quote/_style.scss (hunk 6 above)
Comment (JustinyAhin) [reply assets/sass/05-blocks/quote/_style.scss:2]: Yes.

**Comment by carolinan** [comment]
Closing this in favor of https://github.com/WordPress/twentytwentyone/pull/513.
